### PR TITLE
Add Unit tests, Fix serialization to use json

### PR DESCRIPTION
**Pull request by sagersmith8**
#### Changes:
- Added unit tests
- Fixed serializing a graph to use json
- Fixed deserializing a graph to use json
- Updated requirements for unit tests
- Updated travis to run unit tests
- Broke out duplicate code into a separate function
- Fixed travis to make a directory for the tests to use
##

### CODE REVIEW

File: tests/test_problem_generator.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Tests write graph to file
+        """
+        num_verts = [0, 5, 100]
+        for index, num_vert in enumerate(num_verts):
+            problem_generator.write_graph_to_file(
+                'test{}.json'.format(index),
+                problem_generator.generate_graph(num_vert)
+            )
+
+            # This will need to change once generate graph is done
```
Comment (Kuroshii) [inline tests/test_problem_generator.py]: Maybe just test that the thing you read from the file is the same as what you wrote to the file?

Comment (sagersmith8) [reply tests/test_problem_generator.py]: Noted, I will make that change later tonight

File: tests/test_problem_generator.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            # This will need to change when generate graph is finished
+            self.assertIsNone(
+                problem_generator.generate_graph(num_vert)
+            )
+
+    def test_build_graph(self):
+        """
+        Tests build graph
+        """
+        points = [
```
Comment (Kuroshii) [inline tests/test_problem_generator.py:51]: I don't think this is valid input for `build_graph`;

Edit: never mind, the names are just strange

File: ai_graph_color/problem_generator.py [hunk 3]
```diff
@@ -17,15 +18,21 @@ def write_graph_to_file(out_file, graph):
...
-            ' '.join(
-                map(
-                    lambda x: str(x), adjacency_list
-                )
-            ) + '\n' for adjacency_list in graph
-        )
+        json.dump(graph, out_file, encoding='utf8')
+
+
+def create_file(file_name):
```
Comment (Kuroshii) [inline ai_graph_color/problem_generator.py]: This function seems pretty vaguely named and commented; maybe something like `build_problem_path` or something would be a better name?

Comment (sagersmith8) [reply ai_graph_color/problem_generator.py]: Yes sounds good

**Pull request merged**